### PR TITLE
Construct keys in cxl-ide rootport side

### LINKED ISSUE #142: CXL.memcache Key ordering is not correct in root port side

### PR DESCRIPTION
**Pull request by mxu9**
Fix #142

### CODE REVIEW

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c [hunk 1]
```diff
@@ -143,18 +156,25 @@ bool cxl_setup_ide_stream(void *doe_context, void *spdm_context,
     return false;
   }
   TEEIO_DEBUG((TEEIO_DEBUG_INFO, "key_prog TX - %02x\n", kp_ack_status));
+  TEEIO_DEBUG((TEEIO_DEBUG_INFO, "Print key/iv send to device side:\n"));
+  cxl_dump_key_iv((uint8_t *)tx_key_buffer.key, sizeof(tx_key_buffer.key), (uint8_t *)tx_key_buffer.iv, sizeof(tx_key_buffer.iv));
 
   // Program TX/RX pending keys into Link_Enc_Key_Tx and Link_Enc_Key_Rx registers
   // Program TX/RX IV values
-  revert_copy_by_dw(tx_key_buffer.key, sizeof(tx_key_buffer.key), keys.bytes, sizeof(keys.bytes));
```
Comment (jyao1) [inline teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c:149]: revert_copy_by_dw -> pcie_xxx

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c [hunk 2]
```diff
@@ -143,18 +156,25 @@ bool cxl_setup_ide_stream(void *doe_context, void *spdm_context,
...
+  reordering_keys_in_rp((uint8_t *)tx_key_buffer.key, sizeof(tx_key_buffer.key));
```
Comment (jyao1) [inline teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c:164]: reordering_keys_in_rp -> cxl_xxx

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c [hunk 3]
```diff
@@ -106,6 +115,7 @@ bool cxl_setup_ide_stream(void *doe_context, void *spdm_context,
     TEEIO_DEBUG((TEEIO_DEBUG_ERROR, "libspdm_get_random_number for rx_key_buffer failed.\n"));
     return false;
   }
+  rx_key_buffer.iv[1] = 1;
```
Comment (jyao1) [inline teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c:118]: `iv[1] = 1` ?

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c (hunk 1 above)
Comment (mxu9) [reply teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c:149]: renamed to pcie_construct_rp_keys

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c (hunk 2 above)
Comment (mxu9) [reply teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c]: renamed to cxl_construct_rp_keys

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c (hunk 3 above)
Comment (mxu9) [reply teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c]: It is not needed because default IV is used.

**Pull request merged**